contrib/tools/config-docs-generator/Dockerfile

```diff
@@ -0,0 +1,33 @@
+# Use a specific nightly toolchain for reproducible builds
+FROM rustlang/rust@sha256:04690ffa09cddd358b349272173155319f384e57816614eea0840ec7f9422862
+
+RUN apt-get update -y \
+ && apt-get install -y --no-install-recommends \
+    jq \
+ && rm -rf /var/lib/apt/lists/*
+
+# Set the working directory for building
+WORKDIR /build
+
+# Copy the entire project root to preserve structure
+# Copy from three levels up (project root) to maintain the directory structure
+COPY ../../../ /build
+
+# Pre-build the config-docs-generator binaries during image build
```

Comment:
I think we should update our dockerfiles to minimize root executions. The security implications of root-user dockerfiles kind of depends on the environment the container runs in, but for the most part, I think our dockerfiles could easily support non-root users so its a practice we should probably switch to.

I think here it might be as simple as adding:

```dockerfile
RUN useradd -ms /bin/bash docs-builder
RUN chown docs-builder -R /build
USER docs-builder
```

After the `COPY` instruction.